src/utils/system_utils/src/system_utils.c

```diff
@@ -67,6 +68,52 @@ const char* ADUC_SystemUtils_GetTemporaryPathName()
     return env;
 }
 
+/**
+ * @brief mktemp() implementation
+ *
+ * mktemp() is generally considered unsafe as the temporary names can be easy to guess, and there are
+ * a limited number of combinations.
+ *
+ * However, for unit tests, it can be useful, so it's highly recommended that this method is only
```

Comment:
If putting it in system_util it would eventually get used despite the comments.  Please move this into test_utils instead if only meant for test scenarios.